src/sentry/integrations/messaging.py

```diff
@@ -0,0 +1,218 @@
+            )
+
+        vs = self.identity_view_set
+        return [
+            build_path(operation_slug, view_cls)
+            for (operation_slug, view_cls) in [
+                ("link-identity", vs.link_personal_identity),
+                ("unlink-identity", vs.unlink_personal_identity),
+                ("link-team", vs.link_team_identity),
+                ("unlink-team", vs.unlink_team_identity),
+            ]
+            if view_cls is not None
+        ]
+
+    @abstractmethod
+    def send_incident_alert_notification(
+        self,
+        action: AlertRuleTriggerAction,
+        incident: Incident,
+        metric_value: float,
+        new_status: IncidentStatus,
+        notification_uuid: str | None = None,
+    ) -> bool:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def notify_service_action(self) -> type[IntegrationEventAction] | None:
+        """Define the event action to be added to the global RuleRegistry."""
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def notification_sent(self) -> type[analytics.Event] | None:
+        """Define an analytics event for a notification being sent.
+
+        Development note: This one was singled out purely because MsTeams does it.
+        See `sentry/integrations/slack/analytics.py` for a lot of other events that
+        we might want to make universal to all messaging integrations. Once we have a
+        common set of analytics events, it probably is best to represent them as a
+        compound object (similar to `identity_view_set`) rather than as a bunch of
+        class-level properties.
+        """
+        raise NotImplementedError
+
+    # Requirements for messaging feature per tech spec
+
+    @property
+    def help_view(self) -> type[View] | None:
+        """TODO?"""
+        return None
+
+    @property
+    def support(self) -> type[View] | None:
+        """TODO?"""
+        return None
+
+    @property
+    def documentation(self) -> type[View] | None:
+        """TODO?"""
+        return None
+
+
+class MessagingActionHandler(DefaultActionHandler):
+    def __init__(
+        self,
+        action: AlertRuleTriggerAction,
+        incident: Incident,
+        project: Project,
+        spec: MessagingIntegrationSpec,
+    ):
+        super().__init__(action, incident, project)
+        self._spec = spec
+
+    @property
+    def provider(self) -> str:
+        return self._spec.provider_slug
+
+    def send_alert(
+        self,
+        metric_value: int | float,
+        new_status: IncidentStatus,
+        notification_uuid: str | None = None,
+    ) -> None:
+        success = self._spec.send_incident_alert_notification(
+            self.action, self.incident, metric_value, new_status, notification_uuid
+        )
+        if success:
+            self.record_alert_sent_analytics(self.action.target_identifier, notification_uuid)
```

Comment:
Do we have any place that captures metrics or sentry issues around alert failures? If not, this might be a good thing to add here.

Reply:
I'd be happy to add one. Would we want it to go to `analytics` as `record_alert_sent_analytics` does? Or just `metrics` or `logger`?

Reply:
What's the main use case for `analytics`? I sort of just assumed it was mainly for bizops tracking 😅 

I typically use metrics + optionally logger (depending on the volume)